### PR TITLE
update environments telemetry to add hresults

### PR DESCRIPTION
**Pull request by bbonaby**
## Summary of the pull request

Although we capture the error messaging, we don't capture the HResult error codes. Capturing these will provide us with more information that we can use to filter out noisy data a bit easier.

Changes:

1. Added a `TelemetryResult` class that extracts data from the `ProviderOperationResult` so this doesn't have to be done by all the other telemetry classes.
2. Added an HResult integer property for the `EnvironmentOperationEvent`, `EnvironmentCreationEvent` and the `EnvironmentLaunchEvent` classes. These also now take a `TelemetryResult`  in their constructors to get the error information.

## Validation steps performed

I used the TraceView++ app to confirm that the events are now submitting the HResults, that are received by the operations in their payload.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs [hunk 1]
```diff
@@ -317,18 +319,25 @@ public void Receive(ComputeSystemOperationStartedMessage message)
...
+                data.ComputeSystemOperation,
+                providerId,
+                new TelemetryResult(),
+                data.AdditionalContext);
+
+        TelemetryFactory.Get<ITelemetry>().Log(
+            "Environment_OperationInvoked_Event",
+            LogLevel.Critical,
+            telemetryPayload,
+            relatedActivityId: message.Value.ActivityId);
```
Comment (krschau) [inline tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs:340]: nit: indentation

File: common/TelemetryEvents/TelemetryResult.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public string? DisplayMessage { get; private set; }
+
+    public string? DiagnosticText { get; private set; }
+
+    public TelemetryResult(ProviderOperationResult? result)
+    {
+        UpdateProperties(result);
+    }
+
+    public TelemetryResult(int hResult, string displayMessage, string diagnosticText)
```
Comment (krschau) [inline common/TelemetryEvents/TelemetryResult.cs:27]: This seems like a good candidate to document in https://github.com/microsoft/devhome/tree/main/docs/tools/common. That way we don't forget things, and it encourages reuse.

Comment (bbonaby) [reply common/TelemetryEvents/TelemetryResult.cs:27]: will do. Thanks

File: common/TelemetryEvents/TelemetryResult.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+using Microsoft.Windows.DevHome.SDK;
+
+namespace DevHome.Common.TelemetryEvents;
+
+public class TelemetryResult
+{
+    private const int SuccessHResult = 0;
+
+    private const int InvalidArgHResult = unchecked((int)0x80070057);
```
Comment (sshilov7) [inline common/TelemetryEvents/TelemetryResult.cs:12]: Windows.Win32.Foundation.HRESULT.E_INVALIDARG after adding E_INVALIDARG to NativeMethods.txt.

**Review by sshilov7 — APPROVED** [review]

File: common/TelemetryEvents/TelemetryResult.cs (hunk 2 above)
Comment (bbonaby) [reply common/TelemetryEvents/TelemetryResult.cs:26]: updated.

**Review by krschau — APPROVED** [review]
Hope you don't mind, I pushed a commit with only whitespace changes, so you'll need to pull if you make any other changes.

**Pull request merged**